src/roswire/ros2/node_manager.py

```diff
@@ -23,6 +23,10 @@ class ROS2NodeManager(NodeManager):
     app_instance: 'AppInstance' = attr.ib()
     _state_probe: 'ROS2StateProbe' = attr.ib(init=False)
 
+    def __attrs_post_init__(self) -> None:
+        app_instance = self.app_instance
+        state_probe = self._state_probe
```

Comment:
`self._state_probe` hasn't been defined, so this line will throw an exception. This method should be used to define `self._state_probe`. If the object wasn't frozen, then you could do `self._state_probe = ...`, but since the object is frozen, you'll need to use `object.__setattr__` (see https://github.com/ChrisTimperley/roswire/blob/9e7cffa35693d40bdf31a8ddb757b8ef1b8f6c57/src/roswire/ros2/ros2.py#L32)